### PR TITLE
Added create share / share-request endpoint

### PR DESCRIPTION
**Pull request by azharmalik3**
@mosic

### CODE REVIEW

File: web/views/camera_share_request_view.ex [hunk 1]
```diff
@@ -6,6 +6,10 @@ defmodule EvercamMedia.CameraShareRequestView do
     %{share_requests: render_many(camera_share_requests, __MODULE__, "camera_share_request.json")}
   end
 
+  def render("show.json", %{camera_share_requests: camera_share_request}) do
+      %{share_requests: render_many([camera_share_request], __MODULE__, "camera_share_request.json")}
```
Comment (mosic) [inline web/views/camera_share_request_view.ex]: This should be indented only 2 spaces.

**Comment by mosic** [comment]
@azharmalik3 Nice work. The PR is almost ready to merge, it would just be better to use Ecto Changeset for parameter validation. Consult with @ijunaidfarooq and let me know if you have questions about this.

File: web/models/camera_share_request.ex [hunk 2]
```diff
@@ -33,11 +38,47 @@ defmodule CameraShareRequest do
...
+        rights: rights,
+        message: message,
+        key: UUID.uuid4(:hex)
+      }
+    changeset = changeset(%CameraShareRequest{}, share_request_params)
+    case Repo.insert(changeset) do
+      {:ok, share_request} ->
+        camera_share_request =
+          share_request
+            |> Repo.preload(:camera)
```
Comment (mosic) [inline web/models/camera_share_request.ex]: Line up pipes with `share_request`.

File: web/models/camera_share_request.ex [hunk 3]
```diff
@@ -33,11 +38,47 @@ defmodule CameraShareRequest do
...
     case status do
       "used" -> @status.used
       "cancelled" -> @status.cancelled
       "pending" -> @status.pending
     end
   end
+
+  def create_share_request(camera, email, sharer, rights, message) do
+    share_request_params = %{
+        camera_id: camera.id,
```
Comment (mosic) [inline web/models/camera_share_request.ex]: 2 space indentation.

**Comment by mosic** [comment]
@azharmalik3 You say so, but it's not really the case :)

Checks from validation module should be done in a changeset.

There are also some of the same issues from the other PR: https://github.com/evercam/evercam-media/pull/253

**Comment by mosic** [comment]
Same comments as in https://github.com/evercam/evercam-media/pull/258 apply. In general, don't use nested `if`s, use either `with` or `cond` in those cases.

**Comment by azharmalik3** [comment]
@mosic I have rebased branch on top of master and ready to merge.

**Comment by mosic** [comment]
@azharmalik3 Nice work, rebased into master. I've refactored `CameraShare.validate_rights` and `CameraShareRequest.validate_rights` in https://github.com/evercam/evercam-media/commit/7ff42540c6744815d7d8da7af4af6a13e6549220.

It's now one shared function, doesn't have nested ifs, duplicated error messages or unnecessary arguments. Please use it as a reference for similar code in the future.

**Pull request closed without merging**